sync/clients/sync.py

```diff
@@ -194,6 +235,32 @@ def onboard_workflow(self, workspace_id, job_id: str, project_id: str) -> dict:
             )
         )
 
+    def get_latest_project_config_recommendation(self, project_id: str) -> ProjectConfiguration:
+        latest_recommendation = self.get_latest_project_recommendation(project_id)
+        if latest_recommendation.get("result"):
+            return latest_recommendation["result"][0]["recommendation"]["configuration"]
+        else:
+            return {"error": f"No project recommendation found for Project {project_id}"}
+
+    def get_cluster_definition_and_recommendation(
+        self, project_id: str, cluster_spec_str: str
```

Comment:
Why is `cluster_spec_str` here and in `get_updated_cluster_defintion` a string and not a dict?

Seems weird to me to load json instead of passing a dict directly.

Reply:
I made this parameter a string because my understanding is that the cluster spec would be loaded in the command line as text